### PR TITLE
Authorization * changed work with commands * changed method of creati…

### PR DESCRIPTION
**Pull request by 93artur**
…ng new client * changed class Account to CountsBase * added new variable countNumber, that is used for transfer operation

### CODE REVIEW

**Review by IgorTsalko — CHANGES_REQUESTED** [review]

File: src/main/java/com/arturobank/CountsBase.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    public void changeCount(int countNumber, int count) {
+        counts.replace(countNumber, count);
+    }
+
+    public int createCountNumber(){
+        int max = 9999;
+        int min = 1111;
+        return (int) (Math.random() * ++max) + min;
```
[the remark this answers is not in the document]
Comment (93artur) [reply src/main/java/com/arturobank/CountsBase.java:28]: read about collision. Got it. Resolved

File: src/main/java/com/arturobank/CountsBase.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package com.arturobank;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class CountsBase {
```
[the remark this answers is not in the document]
Comment (93artur) [reply src/main/java/com/arturobank/CountsBase.java:6]: resolved

File: src/main/java/com/arturobank/ClientBase.java [hunk 3]
```diff
@@ -14,11 +15,26 @@ public void setClientsOfBank(Map<String, Client> clientsOfBank) {
         this.clientsOfBank = clientsOfBank;
     }
 
-    public Client getClient(String name) {
+    public Client getClientByCountNumber(int countNumber) {
+        Client client = null;
+        ArrayList<Client> values = new ArrayList<>(clientsOfBank.values());
```
[the remark this answers is not in the document]
Comment (93artur) [reply src/main/java/com/arturobank/ClientBase.java:20]: fixed

File: src/main/java/com/arturobank/ArturoBank.java [hunk 4]
```diff
@@ -89,39 +100,90 @@ public void executeTransaction(BufferedReader reader) throws IOException {
...
+        bank.registerUser("Egor", 2, 2);
+        String userName;
+        int password;
         try (BufferedReader reader = new BufferedReader(new InputStreamReader(System.in))) {
+            System.out.println("Username:");
+            while (true) {
+                userName = reader.readLine();
+                if (userName != null && userName.matches("^[a-zA-Z]+$")) {
+                    if (bank.clientBase.getClientByName(userName) != null) {
+                        bank.myClient = bank.clientBase.getClientByName(userName);
```
[the remark this answers is not in the document]
Comment (93artur) [reply src/main/java/com/arturobank/ArturoBank.java:132]: fixed

File: src/main/java/com/arturobank/ArturoBank.java [hunk 5]
```diff
@@ -29,57 +30,67 @@ public String getCurrentDateTime() {
...
-            if (recipientName != null && recipientName.matches("^[a-zA-Z]+$")) {
-                recipient = clientBase.getClient(recipientName);
-                if (recipient != null) {
-                    break;
-                }
-                System.out.println("There is no client named " + recipientName);
-            } else {
-                System.out.println("Invalid name format");
+            if (myClient.getCountNumber() == countNumber) {
+                System.out.println("You can't choose your account");
```
[the remark this answers is not in the document]
Comment (93artur) [reply src/main/java/com/arturobank/ArturoBank.java:86]: because i cant send money from my account to my account - checking it

**Review by IgorTsalko — APPROVED** [review]

**Pull request merged**